### ISSUE TITLE
iOS: Respect Dark/Light theme settings on CarPlay

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#2487]
There are Auto/Dark/Light settings in OM and a big issue about it here: #749 

At the moment, the CarPlay implementation always follows the car's settings ("auto" mode). It should also respect  "always light" and "always dark" settings.

**Comment by rtsisyk** [comment]
It looks like that the default value is "always dark".

**Comment by biodranik** [comment]
@rtsisyk no. Don't mix different settings here. You're likely speaking about the car's settings when there is "always dark" set in the car's CarPlay:
![image](https://github.com/organicmaps/organicmaps/assets/170263/6adc72b2-08ae-4a33-ac83-4aced8765b75)

I'm speaking about OM's settings, that they should be respected also with CarPlay mode enabled.

**Comment by biodranik** [comment]
After testing in a car, it was impossible to enable Dark mode in CarPlay, even if Always Dark was activated in CarPlay, and Dark theme was selected in OM. Does anyone else observe a similar behavior?

One of the possible proposals @kirylkaveryn @fabwu 

OM Setting | CarPlay Setting | OM Theme          | CarPlay Theme
-----------|-----------------|-------------------|------------------
Auto       | Automatic       |Follows the system |Follows the CarPlay system (may become dark in tunnels)
Light      | Automatic       |Light              |Follows the CarPlay system (may become dark in tunnels)
Dark       | Automatic       |Dark               |Dark
Auto       | Always Dark     |Follows the system |Dark
Light      | Always Dark     |Light              |Dark
Dark       | Always Dark     |Dark               |Dark

**Comment by biodranik** [comment]
Just realized that there are also system Dark/Light theme settings missing in the table above:
- Light
- Dark
- Auto (means scheduled one, either from sunrise to sunset, or at a custom time)

The same situation is on Android.

@AndrewShkrob, what are the possible settings on Android Auto? I saw that there is "Enable Dark Mode in Maps" setting, can we reuse it?

**Comment by biodranik** [comment]
To make it easier, we need to add a _separate_ theme setting into the CarPlay:
- Auto (or Follow the system theme)
- (Force) Light
- (Force) Dark

Users complain that on some cars, when the headlights are turned on, the vehicle switches CarPlay to dark mode. Google Maps has a CarPlay setting to force Light mode.

CC @kirylkaveryn @fabwu
